practices/perform-static-code-analysis.md

```diff
@@ -1,115 +1,71 @@
-# Perform Static Code Analysis
+# Perform Automated Code Analysis
 
-Performing static code analysis involves using automated tools to review and scan the codebase for potential issues, ensuring adherence to quality standards and best practices.
-These tools help detect issues early in development, integrating with version control systems, IDEs, and CI/CD pipelines to enhance productivity.
-Static code analysis is valuable for spotting code smells, basic security vulnerabilities, performance bottlenecks, and analyzing dependencies for better modularity.
+Manually spotting every potential bug, style inconsistency, or design flaw is a tall order — and often a slow one. Automated code analysis brings speed and consistency by having tools (both traditional static analyzers and modern AI-powered assistants) scan code as you work. These tools can highlight security vulnerabilities, style discrepancies, dependency risks, and even suggest or apply fixes in real time.
 
-## Nuance
+Some popular tools include:
+- [ESLint](https://eslint.org/docs/latest/use/getting-started) - General-purpose static analysis for JavaScript/TypeScript
+- [Prettier](https://prettier.io/docs/integrating-with-linters) - Automated code formatting for JavaScript/TypeScript
+- [SonarQube](https://www.sonarsource.com/sem/products/sonarqube/downloads/) - Multi-language static analysis with some AI-powered features
+- [Semgrep](https://github.com/semgrep/semgrep) - Multi-language static and semantic analysis, with AI-assisted rule generation
+- [Claude Code](https://www.anthropic.com/claude) - AI-powered code review, style enforcement, and bug detection
+- Self-hosted LLMs - Using tools like Ollama or LM Studio to run open-source AI models locally
 
-### Common Misconceptions about Static Code Analysis
+## Who It’s For & Why
 
-A common misconception is that static code analysis can catch all possible issues in a codebase.
-While these tools are powerful for identifying code smells, basic security vulnerabilities, and performance bottlenecks, they are not foolproof.
-They may miss more nuanced or context-specific problems, and sometimes flag good code as problematic.
-Developers should not solely rely on these tools but use them as part of a broader quality assurance strategy.
+- You are a developer and would be need feedback on potential bugs, design issues, and style mismatches without waiting for a review cycle.
+- You are a QA Engineer and you need to identify high risk areas earlier in the development process, so you can identify the best use of your often limited testing time ensuring greater code quality.
+- You are a Tech Lead or Manager and you need to ensure consistent code quality across a team without increasing review overhead.
 
-### Importance of Developer Judgment
+## Goals, Metrics & Signals
```

Comment:
I apologize, but I've moved the goal posts ever so slightly on this section. Can you review the updated [template](https://github.com/pragmint/open-practices/blob/main/templates/new-practice.md) and [example practice](https://github.com/pragmint/open-practices/blob/main/practices/migrate-to-monorepo.md) and make tweaks to follow that new format?

Reply:
Also, if after reviewing, you have differing opinions, feel free to share those [here](https://github.com/pragmint/open-practices/pull/52) even though the PR is closed.